File: SRC/Service/Saras.eMarking/Saras.eMarking.Domain/ViewModels/UserManagementModel.cs
using System;
using System.Collections.Generic;

namespace Saras.eMarking.Domain.ViewModels
{
    public class GetAllApplicationUsersModel
    {
        public GetAllApplicationUsersModel()
        {
            getAllUsersModel = new List<GetAllUsersModel>();
            userscount = new Userscount();
        }
        public List<GetAllUsersModel> getAllUsersModel { get; set; }
        public Userscount userscount { get; set; }

    }
    public class GetAllUsersModel : GetAllUnMappedUsersModel
    {
        public bool? isDisable { get; set; }
        public int ROWCOUNT { get; set; }
        public long CuurentloggedinUserId { get; set; }
        public string Name { get; set; }
        public string LoginName { get;set; }

    }

    public class GetAllMappedUsersModel
    {
        public string Name { get; set; }
        public long ProjectuserroleID { get; set; }
        public string LoginName { get; set; }
        public string RoleName { get; set; }
        public string SchooolName { get; set; }
        public string NRIC { get; set; }
        public string Phone { get; set; }
        public bool? Status { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string RoleCode { get; set; }
        public Boolean IsActive { get; set; }
        public long UserId { get; set; }
        public Boolean IsS1Enabled { get; set; }
        public Boolean IsLiveMarkingEnabled { get; set; }
        public long CurrentuserroleID { get; set; }
    }

    public class MappedUserscount
    {
        public int? RoleCounts { get; set; }
        public int? FilterUsersCount { get; set; }
        public string RoleCode { get; set; }
        public string RoleName { get; set; }
        public long? UnMappedUserCount { get; set; }
        public Boolean AOExist { get; set; }
    }
    public class MappedUsersList
    {
        public List<GetAllMappedUsersModel> allmappeduserlist { get; set; }
        public List<GetAllMappedUsersModel> OnlyAOresult { get; set; }
        public List<MappedUserscount> mappedusercount { get; set; }
        public int UnMappedUserscount { get; set; }
        public string currentloginrolecode { get; set; }
        public int FilterUserCount { get; set; }
    }

    public class UNMappedUsersList
    {
        public List<GetAllUnMappedUsersModel> allunmappeduserlist { get; set; }
        public MappedUserscount unmappedusercount { get; set; }
        public string currentloginrolecode { get; set; }
        public Boolean AoCount { get; set; }
    }

    public class GetAllUnMappedUsersModel
    {
        public long UserId { get; set; }
        public int ApplicationLevel { get; set; }
        public string Name { get; set; }
        public string LoginName { get; set; }
        public string RoleName { get; set; }
        public string RoleCode { get; set; }
        public string SchooolName { get; set; }
        public string NRIC { get; set; }
        public string Phone { get; set; }
        public bool? Status { get; set; }
        public bool? isactive { get; set; }
        public bool? isblocked { get; set; }
        public long totalusercount { get; set; }

    }

    public class Userscount
    {
        public long Totalusers { get; set; }
        public long Activeusers { get; set; }
        public long InActiveusers { get; set; }
        public long Blockedusers { get; set; }
        public long ApplicationLevelOfLoginUserID { get; set; }
    }
    public class GetCreateEditUserModel
    {
        public GetCreateEditUserModel()
        {
            Examlevels = new List<ExamLevels>();
            schools = new List<SchoolDetails>();
        }
        public long UserId { get; set; }
        public string Username { get; set; }
        public string Loginname { get; set; }
        public string Nric { get; set; }
        public string PhoneNum { get; set; }
        public string RoleCode { get; set; }
        public string RoleName { get; set; }
        public string SchooolCode { get; set; }
        public string SchoolName { get; set; }
        public List<RoleDetails> roles { get; set; }
        public List<SchoolDetails> schools { get; set; }
        public List<ExamLevels> Examlevels { get; set; }
        public Boolean checkdataexist { get; set; }
        public string selectedroleinfo { get; set; }
    }

    public class CreateEditUser
    {
        public long UserId { get; set; }
        public string Username { get; set; }
        public string Loginname { get; set; }
        public string firstName { get; set; }
        public string lastName { get; set; }
        public string RoleCode { get; set; }
        public string SchooolCode { get; set; }
        public string Nric { get; set; }
        public string PhoneNum { get; set; }
        public string EmailId { get; set; }
        public string RoleName { get; set; }
        public string SchoolName { get; set; }
        public string ProjectRoleName { get; set; }
        public List<ExamLevels> Examlevels { get; set; }
    }

    public class RoleDetails
    {
        public long RoleID { get; set; }
        public string RoleName { get; set; }
        public string RoleCode { get; set; }
        public short? ApplicationLevel { get; set; }

    }

    public class SchoolDetails
    {
        public long SchoolID { get; set; }
        public string SchoolName { get; set; }
        public string SchoolCode { get; set; }
    }

    public class RoleSchooldetails
    {
        public List<RoleDetails> roles { get; set; }
        public List<SchoolDetails> schools { get; set; }
    }

    public class ExamLevels
    {
        public long ExamLevelID { get; set; }
        public string ExamLevelCode { get; set; }
        public string ExamLevelName { get; set; }
        public bool isselected { get; set; }
    }

    public class ImportUsers
    {
        public string FirstName { get; set; }
        public string LoginName { get; set; }
        public string NRIC { get; set; }
        public string Role { get; set; }
        public string School { get; set; }
        public string Status { get; set; }
    }

    public class SearchFilterModel
    {
        public string SearchText { get; set; }
        public string SchoolCode { get; set; }
        public string RoleCode { get; set; }
        public string Status { get; set; }
        public int PageNo { get; set; }
        public int PageSize { get; set; }
        public string SortField { get; set; }
        public int SortOrder { get; set; }
        public Boolean UnmappedUserCountbit { get; set; }

        public int navigate { get; set; }
    }

    public class UserDetails
    {
        public UserDetails()
        {

            Error = new List<string>();

        }
        public string FirstName { get; set; }
        public string LoginName { get; set; }
        public string NRIC { get; set; }
        public string RoleCode { get; set; }
        public string SchoolName { get; set; }
        public string SchoolCode { get; set; }
        public bool Status { get; set; }
        public string Password { get; set; }
        public List<string> Error { get; set; }
        public string PhoneNumber { get; set; }
    }

    public class UserCreations
    {
        public string status { get; set; }
        public List<UserDetails> users { get; set; }
    }

    public class PassPharseModel
    {
        public string PassPharseCode { get; set; }
        public bool? IsActive { get; set; }
        public long? ModifiedBy { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public string ModifiedLoginId { get; set; }
    }
    public class modelName
    {
        public long UserId { get; set; }
        public string loginName { get; set; }
    }

}

File: SRC/Manifest/Saras.eMarkingClient/src/app/model/Global/UserManagement/UserManagementModel.ts
export class GetAllApplicationUsersModel {
    getAllUsersModel!: GetAllUsersModel[];
    userscount!: Userscount;
}

export class GetAllUsersModel {
    UserId!: number;
    ApplicationLevel!: number;
    Name!: string;
    LoginName!: string;
    RoleName!: string;
    SchooolName!: string;
    NRIC!: string;
    Phone!: number;
    Status!: boolean;
    isdeleted!: boolean;
    isactive!: boolean;
    isblocked!: boolean;
    RoleCode!: string;
    isDisable!: boolean;
    ROWCOUNT!: number;
    CuurentloggedinUserId!:number;
    selected!: boolean;
}

export class Userscount {
    Totalusers!: number;
    Activeusers!: number;
    InActiveusers!: number;
    Blockedusers!: number;
    ApplicationLevelOfLoginUserID!: number;
}

export class GetCreateEditUserModel {
    UserId!: number;
    Username!: string;
    Loginname!: string;
    Nric!: string;
    PhoneNum!: string;
    RoleCode!: string;
    RoleName!: string;
    SchooolCode!: string;
    SchoolName!: string;
  selectedroleinfo!: string;
  checkdataexist!: boolean;
    roles!: RoleDetails[];
    schools!: SchoolDetails[];
    Examlevels!: ExamLevels[];
}

export class ExamLevels {
    ExamLevelID!: number;
    ExamLevelCode!: string;
    ExamLevelName!: string;
    isselected!: boolean;
}

export class CreateEditUser {
    UserId!: number;
    Username!: string;
    Loginname!: string;
    SchooolCode!: string;
    Nric!: string;
    PhoneNum!: string;
    EmailId!: string;
    IsBlock!: string;
    ForgotCount!: number;
    LoginCount!: number;
    RoleCode!: string;
    RoleName!: string;
    firstName!: string;
    lastName!: string;
    SchoolName!: string;
    ProjectRoleName!:string;
    Examlevels!: ExamLevels[];
}

export class RoleDetails {
    RoleID!: number;
    RoleName!: string;
    RoleCode!: string;
    IsRoleSelected!: boolean;
}

export class SchoolDetails {
    SchoolID!: number;
    SchoolName!: string;
    SchoolCode!: string;
    IsSchoolSelected!: boolean;
}

export class RoleSchooldetails {
    roles!: RoleDetails[];
    schools!: SchoolDetails[];
}

export class ImportUsers {
    FirstName!: string;
    LoginName!: string;
    NRIC!: string;
    Role!: string;
    School!: string;
    Status!: boolean;
}

export class SearchFilterModel {
    SearchText!: string;
    SchoolCode: string = "";
    RoleCode: string = "";
    Status: string = "";
    PageNo!: number;
  PageSize!: number;
  SortField: string = "";
  SortOrder: number = 0;
  navigate!:number
}

export class PassPhrase {
    Id!: number;
    PassPharseCode!: string;
    IsActive!: boolean;
    IsDeleted!: boolean;
    CreatedBy!: number;
    CreatedDate!: string;
    ModifiedBy!: number;
    ModifiedDate!: string;
}

export class UserDetails {
    FirstName!: string;
    LoginName!: string;
    NRIC!: string;
    RoleCode!: string;
    SchoolName!: string;
    SchoolCode!: string;
    Status!: boolean;
    Password!: string;
    Error!: any[];

}

export class UserCreations {
    status!: string
    users!: UserDetails[];
}
